### ISSUE TITLE
do not pass tests on julia 1.9

### ISSUE THREAD

**Issue by rveltz** [nep-pack/NonlinearEigenproblems.jl#265]
Hi,

I am having errors on 1.9

Here is what I get on osx. Am I doing something wrong?

Thank you

```julia
Running test AAAeigs (1 / 52)
AAAeigs: Error During Test at /Users/rveltz/.julia/packages/NonlinearEigenproblems/S9WLC/test/AAAeigs.jl:5
  Got exception outside of a @test
  UndefVarError: `umf_ctrl` not defined
  Stacktrace:
    [1] getproperty
      @ ./Base.jl:31 [inlined]
    [2] with_umfpack_refinements(f::NonlinearEigenproblems.LinSolvers.var"#2#3"{FactorizeLinSolver{LU{ComplexF64, Matrix{ComplexF64}, Vector{Int64}}}, Vector{ComplexF64}}, refinements::Int64)
      @ NonlinearEigenproblems.LinSolvers ~/.julia/packages/NonlinearEigenproblems/S9WLC/src/LinSolvers.jl:137
    [3] #lin_solve#1
      @ ~/.julia/packages/NonlinearEigenproblems/S9WLC/src/LinSolvers.jl:131 [inlined]
    [4] lin_solve(solver::FactorizeLinSolver{LU{ComplexF64, Matrix{ComplexF64}, Vector{Int64}}}, x::Vector{ComplexF64})
      @ NonlinearEigenproblems.LinSolvers ~/.julia/packages/NonlinearEigenproblems/S9WLC/src/LinSolvers.jl:130
    [5] AAAeigs(::Type{ComplexF64}, nep::DEP{Float64, Matrix{Float64}}, Z::Vector{ComplexF64}; logger::Int64, mmax::Int64, neigs::Int64, maxit::Int64, shifts::Vector{ComplexF64}, linsolvercreator::FactorizeLinSolverCreator{Number, Any}, tol::Float64, tol_appr::Float64, v0::Vector{ComplexF64}, errmeasure::ResidualErrmeasure, weighted::Bool, cleanup_appr::Bool, tol_cln::Float64, return_details::Bool, check_error_every::Int64, inner_logger::Int64)
```

where

```julia
pkg> st NonlinearEigenproblems
Status `~/.julia/environments/scratch/Project.toml`
  [067cfa3b] NonlinearEigenproblems v1.1.0
```

**Comment by jarlebring** [comment]
Error originates from when we control number of refinements in umfpack:

https://github.com/nep-pack/NonlinearEigenproblems.jl/blob/6d91752393adbe26e17a1fedfddc5dbcec71304c/src/LinSolvers.jl#L136-L145

This is due to a change in `SuiteSparse.UMFPACK` to remove the use of global variables.  https://github.com/JuliaSparse/SparseArrays.jl/issues/127 . The change in `SuiteSparse.UMFPACK` makes sense.  To be compatible with 1.9:  instead of setting `umf_ctrl`, we need to set the `control` in the `UmfpackLU`-object.

**Comment by jarlebring** [comment]
Closed by 7bd3f7954823aa5685f4547b8821089c18d9dd85 . Should work in new nep-pack version.
